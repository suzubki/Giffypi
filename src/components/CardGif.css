.card-margin{
    padding: 2rem;
}

.card_image--margin{
    margin: 2px auto 0 auto;
}