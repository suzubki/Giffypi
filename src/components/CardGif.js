import React from "react";
import PropTypes from 'prop-types'

// React slick (carousel)
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";

// Reactstrap
import { Card } from "reactstrap";

// CSS
import "./CardGif.css";

export const CardGif = ({ objectData }) => {

  let { data, loading } = objectData

  // Slick settings
  const settings = {
    infinite: true,
    autoplay: true,
    speed: 2000,
    autoplaySpeed: 2000,
    slidesToShow: 4,
    slidesToScroll: 1,
    centerMode: true,
    centerPadding: "10px",
    initialSlide: 0,
    pauseOnHover: true,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 1,
          infinite: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  };

  // Render card gif
  return (
    <>
      <Slider {...settings}>
        {
        loading && data.map((gif) => (
            <Card key={gif.id}>
              <img
                alt="Card cap"
                className="card_image--margin"
                src={gif.images.original.url}
                width="90%"
              />
            </Card>
        ))}
      </Slider>
    </>
  );
};


CardGif.propTypes = {
  objectData: PropTypes.object.isRequired
}

CardGif.defaultProps = {
  objectData: {data: [], loading: Boolean}
}

export default CardGif;
