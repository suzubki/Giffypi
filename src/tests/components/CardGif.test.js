import React from 'react';
import {shallow} from 'enzyme';
import CardGif from '../../components/CardGif';


describe('DeberÃ­a mostrar solo el wrapper', () => { 
    // test('mostrando el wrapper', () => { 
    //     const newOb = []

    //     const wrapper = shallow( <CardGif objectData={newOb} /> )

    //     expect(wrapper).toMatchSnapshot()

    //  })
    
 })