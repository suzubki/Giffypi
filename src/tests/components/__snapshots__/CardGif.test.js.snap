// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deber√≠a mostrar solo el wrapper mostrando el wrapper 1`] = `
<Fragment>
  <Slider
    autoplay={true}
    autoplaySpeed={2000}
    centerMode={true}
    centerPadding="10px"
    infinite={true}
    initialSlide={0}
    pauseOnHover={true}
    responsive={
      Array [
        Object {
          "breakpoint": 1024,
          "settings": Object {
            "infinite": true,
            "slidesToScroll": 1,
            "slidesToShow": 3,
          },
        },
        Object {
          "breakpoint": 600,
          "settings": Object {
            "initialSlide": 2,
            "slidesToScroll": 1,
            "slidesToShow": 2,
          },
        },
        Object {
          "breakpoint": 480,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
      ]
    }
    slidesToScroll={1}
    slidesToShow={4}
    speed={2000}
  />
</Fragment>
`;
